### ISSUE TITLE
Service Worker: Disable auto update

### ISSUE THREAD

**Issue by SvenBudak** [angular/angular#32883]
Hi! Currently i am trying to build an app for a customer that want, that the service worker updates every night at 24 (midnight). I wrote a simple service, but its useless because i cant deactivate at the moment the automatic update:

```
import {Injectable} from '@angular/core';
import {timer} from 'rxjs';
import {SwUpdate} from '@angular/service-worker';

@Injectable({
    providedIn: 'root'
})
export class UpdateApplicationService {

    constructor(private swUpdate: SwUpdate) {
    }

    updateAt(hour = 24, min = 0, sec = 0) {
        const future = new Date() as any;
        future.setHours(hour, min, sec, 0);
        const now = new Date() as any;
        const seconds = future - now;
        const obs = timer(seconds, ((24 * 60) * 60) * 1000);
        obs.subscribe(() => this.update());
    }

    async update() {
        if (this.swUpdate.isEnabled) {
            await this.swUpdate.activateUpdate();
            console.log('Updated');
            window.location.reload();
        } else {
            console.log('ServiceWorker not enabled');
        }
    }

}
```
Is a feature planed soon, to deactivate the automatic checks?

**Comment by gkalpak** [comment]
What exactly is it that you want to avoid? Updating is a multi-step process (downloading the update, activating it for a page, reloading the page). Which part are you trying to prevent (and why, if I may ask)?

**Comment by SvenBudak** [comment]
The PWA i create, will be used world wide, also in countries where they have very slow internet connection. Furthermore, shops use over the day the app to show customers some informations about products. The owner dont want that it update meanwhile they have meetings and use the app etc...

An Update can import in worst case about 5 videos which each of them can have 200 or more megabytes.

They want that the app update at midnight (local time) with a short progressbar that show the update status. (but i am not sure that this is possible currently with current serviceworker version in angular)

**Comment by gkalpak** [comment]
I see. The update happens in the background (and the app is only notified once everything has been downloaded and is ready to use). So, the slow connection shouldn't be an issue.

Currently, it is not possible to disable updates in the background (nor show a progress bar). Note, of course, that an existing page will not receive the update until it is reloaded, but new pages will always get the latest available version.

Also, keep in mind that it is not guaranteed that a `setTimeout()`-scheduled task will be triggered at the specified time or even trigger at all (e.g. if the user closes the page/browser or powers off the device). So, it is not a reliable way to ensure something will happen at a given time.

##
The bottom line is that `@angular/service-worker` does not support your requirements, I'm afraid. You might need to look at another option (e.g. workbox) that allows more customization. Alternatively, you could choose to modify the `ngsw-worker.js` script and disable automatic checks for updates (but this is not an officially supported approach :grin:).

**Comment by SvenBudak** [comment]
Thats really sad... ^^ but thanks alot for the fast reply! you think this feature will come any time in future?

**Comment by gkalpak** [comment]
No specific plans (but we won't stop anyone that wants to submit a PR :wink:).

**Comment by gkalpak** [comment]
As mentioned in my previous comments, this doesn't sound like something we would want to support in `@angular/service-worker`. Given that and the low demand for the feature, I am going to close this.
